src/main/scala/org/broadinstitute/dsde/workbench/sam/backgroundProcess/CryptominingSubscriber.scala

```diff
@@ -0,0 +1,47 @@
+package org.broadinstitute.dsde.workbench.sam
+package backgroundProcess
+
+import cats.effect.{ContextShift, IO}
+import fs2.Stream
+import io.chrisdavenport.log4cats.StructuredLogger
+import io.circe.Decoder
+import org.broadinstitute.dsde.workbench.google2.{Event, GoogleSubscriber}
+import org.broadinstitute.dsde.workbench.model.WorkbenchUserId
+import org.broadinstitute.dsde.workbench.sam.service.UserService
+import org.broadinstitute.dsde.workbench.sam.util.SamRequestContext
+
+// Each instance of Sam will run a subscriber, since they all subscribe to the same topic via same subscription,
+// messages will be distributed among subscribers instead of distributed to all of them, which is what we want.
+class CryptominingSubscriber(subscriber: GoogleSubscriber[IO, CryptominingUserMessage],
+                             userService: UserService
+                            )(implicit logger: StructuredLogger[IO], cs: ContextShift[IO]) {
+  val process: Stream[IO, Unit] = subscriber.messages
+    .parEvalMapUnordered(2)(messageHandler) // concurrency is low since we don't need high performance in disabling cryptomining users.
```

Comment:
why not `evalMap` and forget the question of concurrency? There are other sam instances too so this is never single threaded anyway.

Reply:
I made it 2 just so if in the future there's need to increase concurrency, people don't have to figure out how to use `parEvalMapUnordered`...happy to change this to no concurrency